### ISSUE TITLE
[Bug report]: Input fields background color in dark mode

### ISSUE THREAD

**Issue by luistabotelho** [scania-digital-design-system/tegel#852]
### Requirements before reporting

- [X] No duplicated [issue](https://github.com/scania-digital-design-system/tegel/issues) reported.
- [X] I have checked the latest version if the bug exist there. See all available packages at [npmJS.com](https://www.npmjs.com/package/@scania/tegel)
- [X] I have followed the installation guide.

### Package versions

@scania/tegel-angular-17: 1.18.0

### Browser

Edge (Chromium)

### Framework

Angular

### Version

Angular 18

### Reproduction steps

1. Add a modal with dark mode implemented
2. Add an input to modal (Ex.: TdsTextField)
3. Select appropriate variant for input component to be visible
4. Switch UI between light and dark mode

### Screenshots

![image](https://github.com/user-attachments/assets/f1ddb59c-b835-469e-8954-c5c389e2a3fd)
![image](https://github.com/user-attachments/assets/553bc699-fdef-480c-b60f-eda66ec34a57)

### Expected behaviour

Input fields should be visible against background in both light and dark mode. (background color should be different). Currently with the components in their primary variant they are visible in light mode, but have the same background in dark mode. Switching them to their secondary variant makes them visible in dark mode, but then they have the same background color in light mode, becoming invisible.

In the example pictures the "Distributor Id" input and the TdsBlock around the radio buttons have a variant of secondary, while the rest has a variant of primary. Observe how in light mode the primary ones are visible, while in dark mode the secondary ones are visible.

### Contact information

luis.felipe.botelho@scania.com

**Comment by NOMADE55** [comment]
Hey! First time contributor here. 
I was looking for a way to fix your issue with the colors here, but it seems is not a bug in itself.

Input and Textarea share the same background color with Modal in dark mode.
This most likely should be addressed in design, or by switching dynamically `modeVariant` to secondary for inputs within a modal in dark mode.

**Comment by luistabotelho** [comment]
Indeed it seems to be a problem in design. There is another instance of the problem, if you combine a folder tab and a block (to give the effect of the content being in the tab, it matches colors in light mode, but in dark mode the block disappears.

![image](https://github.com/user-attachments/assets/d4509b4f-03ad-4109-8468-48919840d75f)
![image](https://github.com/user-attachments/assets/3b62fe25-74d0-42c0-a22c-268d17037001)

Switching to mode secondary inverts the effect:

![image](https://github.com/user-attachments/assets/a3bdd62a-4a3b-4b95-8dd5-ed2d8dc449ee)
![image](https://github.com/user-attachments/assets/77b1b29f-cb9a-4512-b955-0afb75bca0cb)

**Comment by mjukiscissi** [comment]
@luistabotelho this issue has been parked as it is blocked by another major initative in our backlog. We will close this issue for now and let you know when we have a fix for it. Reach out if you have any questions!
